src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,38 @@
 package mate.academy;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final int WAIT_MILLISECONDS = 1000;
     private final Executor executor;
+    private final Map<String, UserData> cache = new HashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+
+        return CompletableFuture.supplyAsync(() -> {
+            wait(WAIT_MILLISECONDS);
+            UserData userData = new UserData(userId, "Details for " + userId);
+            cache.put(userId, userData);
+            return userData;
+        }, executor);
+    }
+
+    private void wait(int milliseconds) {
```

Comment:
The method 'wait' is a bit misleading since it shadows the 'Object.wait' method. It's better to use a different method name to avoid confusion. Additionally, using 'Thread.sleep' to simulate a delay is not a good practice in an asynchronous context as it blocks the thread. Consider using a scheduled executor or CompletableFuture's delay methods if available in your Java version.